src/kernel/interrupts.h

```diff
@@ -2,29 +2,34 @@
 
 #include "interrupt.h"
 
+extern volatile u32* syscall_return;
+
 __attribute__((naked)) void common_interrupt_stub()
 {
   asm volatile(
       "pusha \n"          // Pushes edi, esi, ebp, esp, ebx, edx, ecx, eax
-      "mov %ds, %ax \n"   // Set lower 16-bits of eax to ds (data segment selector)
-      "push %eax \n"      // Save the ds
-      "push %esp \n"      // Save the pointer to the register_info struct on the stack
-      "mov $0x10, %ax \n" // Load the kernel data segment descriptor so we can access kernel memory
-      "mov %ax, %ds \n"
-      "mov %ax, %es \n"
-      "mov %ax, %fs \n"
-      "mov %ax, %gs \n"
+      "mov %%esp, %%eax \n"
+      "add $0x1c, %%eax \n"
+      "mov %%eax, (%0) \n" // save the location of where eax is on the stack
```

Comment:
Please make this comment a bit more descriptive, like "Save the address of `eax` on the stack so we can write the syscall return value there". 

Also, does any of this do anything destructive if the given interrupt is not a syscall or the it's a void syscall?

Reply:
Sure, I will fix this and the other suggestions.

This is non-destructive. If you don't write to the pointer eax is unchanged.

Reply:
Ok, cool, thanks!